sdks/python/apache_beam/ml/inference/sklearn_inference.py

```diff
@@ -83,14 +83,21 @@ def _convert_to_result(
     ]
   return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
 
+def _default_numpy_inference_fn(
+  model: BaseEstimator, batch: Sequence[numpy.ndarray], inference_args: Optional[Dict[str, Any]] = None
+) -> Any:
+    # vectorize data for better performance
+    vectorized_batch = numpy.stack(batch, axis=0)
+    return model.predict(vectorized_batch)
 
 class SklearnModelHandlerNumpy(ModelHandler[numpy.ndarray,
                                             PredictionResult,
                                             BaseEstimator]):
   def __init__(
       self,
       model_uri: str,
-      model_file_type: ModelFileType = ModelFileType.PICKLE):
+      model_file_type: ModelFileType = ModelFileType.PICKLE,
+      inference_fn: Optional[Callable[[BaseEstimator, Sequence[numpy.ndarray], Optional[Dict[str, Any]]], Any]] = _default_numpy_inference_fn):
```

Comment:
This is getting unwieldy. Wondering if there's a better way to do this. Could we at least refactor the type for `inference_args: Optional[Dict[str, Any]]`?

Reply:
Would you just want it stepped back to `Optional[Any]` for the sake of being concise?

Reply:
Circled around to defining a custom type to make this look less verbose in function signatures while still providing solid typing guarantees.